### ISSUE TITLE
ccrewrite fails to read assembly that has non-escaped comma in nested type names

### ISSUE THREAD

**Issue by SergeyTeplyakov** [Microsoft/CodeContracts#340]
Related to the #186.

If old version of ccrewrite could generate type names that with non-espcaped comman inside when working with Roslyn-generated code.

It means, that instead of `SomeGeneric+Nested<Int32\,String>` it can generate `SomeGeneric+Nested<Int32,String>`. But actually both cases are correct and metadata reader should deal with both of them smoothly.

**Comment by tom-englert** [comment]
@SergeyTeplyakov with this fix we get crashes in ccrewrite for generic types:

the class with it's attribute looks something like this:
```
[Export(typeof(IEntityToDtoMapping<TestOrder, Modules.TestOrder.ServerAccess.TestOrder, IPersistenceContext>))]
private class Mapping : EntityToDtoMapping<TestOrder, Modules.TestOrder.ServerAccess.TestOrder, IPersistenceContext>
{
```
which will generate a type name like this:
```

**Comment by SergeyTeplyakov** [comment]
Could you provide a small repo solution that I can try? I can revert the change if solution would be complicated...

**Comment by tom-englert** [comment]
I have a fix already in #445, just looking how to add a test for this scenario. Did you have a test for the original issue?

**Comment by SergeyTeplyakov** [comment]
Yep, I've faced an issue but that was not a critical and more artificial case. I'll role this change back.

**Comment by SergeyTeplyakov** [comment]
It seems that you fix it in your fork. If you'll create PR I'll merge it.

**Comment by tom-englert** [comment]
#445 is a pull request :smirk:

**Comment by SergeyTeplyakov** [comment]
Oops! Thought it is an old one. Will merge once appveyor will be finished.

**Comment by tom-englert** [comment]
After adding a test I could not find the scenario described here - instead with the code introduced in #341 it was not able to read it at all, so I have reverted the complete fix in PR #446
